fluent-kafka-streams-tests/src/main/java/com/bakdata/fluent_kafka_streams_tests/TestTopology.java

```diff
@@ -117,15 +118,30 @@ public class TestTopology<DefaultK, DefaultV> {
      * Used by wither methods.
      */
     protected TestTopology(final Function<? super Properties, ? extends Topology> topologyFactory,
-            final Map<?, ?> properties, final Serde<DefaultK> defaultKeySerde,
+            final Function<? super String, ? extends Map<?, ?>> propertiesFactory,
+            final Serde<DefaultK> defaultKeySerde,
             final Serde<DefaultV> defaultValueSerde, final SchemaRegistryMock schemaRegistry) {
         this.schemaRegistry = schemaRegistry;
         this.topologyFactory = topologyFactory;
-        this.properties.putAll(properties);
+        this.propertiesFactory = propertiesFactory;
         this.defaultKeySerde = defaultKeySerde;
         this.defaultValueSerde = defaultValueSerde;
     }
 
+    /**
+     * <p>Create a new  for your topology under test.</p>
```

Comment:
I think something is missing in the text here?
```suggestion
     * <p>Create a new {@link TestTopology} for your topology under test.</p>
```